spec/v2/providers/pubsub.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { expect } from 'chai';
+
+import { CloudEvent } from '../../../src/v2/base';
+import * as options from '../../../src/v2/options';
+import * as pubsub from '../../../src/v2/providers/pubsub';
+
+const EVENT_TRIGGER = {
+  eventType: 'google.cloud.pubsub.topic.v1.messagePublished',
+  resource: 'projects/aProject/topics/topic',
+};
+
+describe('onMessagePublished', () => {
+  beforeEach(() => {
+    options.setGlobalOptions({});
+    process.env.GCLOUD_PROJECT = 'aProject';
+  });
+
+  afterEach(() => {
+    delete process.env.GCLOUD_PROJECT;
+  });
+
+  it('should return a minimal trigger with appropriate values', () => {
+    const result = pubsub.onMessagePublished('topic', () => 42);
+    expect(result.__trigger).to.deep.equal({
+      apiVersion: 2,
+      platform: 'gcfv2',
+      eventTrigger: EVENT_TRIGGER,
+      labels: {},
+    });
+  });
+
+  it('should create a complex trigger with appropraite values', () => {
+    const result = pubsub.onMessagePublished(
+      {
```

Comment:
I wonder if we should extract out a common "kitchen sink" trigger that can be used across tests...

```ts
import { FULL_OPTIONS, FULL_TRIGGER } from "./helpers";

pubsub.onMessagePublished({ ...FULL_OPTIONS, topic: 'topic' });
expect(result.__trigger).to.deep.equal({ ...FULL_TRIGGER, eventTrigger: EVENT_TRIGGER });
```